src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,34 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate localDateFrom = LocalDate.parse(dateFrom, FORMATTER);
+        LocalDate localDateTo = LocalDate.parse(dateTo, FORMATTER);
+
+        StringBuilder sb = new StringBuilder("Report for period " + dateFrom + " - " + dateTo);
+
+        for (String name : names) {
+            int totalSalary = 0;
+            for (String intel : data) {
+                String[] splitIntel = intel.split(" ");
+                LocalDate currentDate = LocalDate.parse(splitIntel[0], FORMATTER);
+                if (name.equals(splitIntel[1])
+                        && !currentDate.isBefore(localDateFrom)
+                        && !currentDate.isAfter(localDateTo)) {
+                    totalSalary += Integer.parseInt(splitIntel[2])
+                            * Integer.parseInt(splitIntel[3]);
```

Comment:
Please, don’t use magic numbers

Reply:
Okay, how would I declare the cells of String array?